### ISSUE TITLE
Android app doesn't save the last edit

### ISSUE THREAD

**Issue by FireDragonRider** [laurent22/joplin#5129]
Android app doesn't save the note's last edit (most often couple of last words)

## Environment

Joplin version: 2.1.3
Platform: Android
OS specifics: EMUI 10.0.0

## Steps to reproduce

1. create a new note
2. write something
3. exit editor, last edit still there
4. go to notebook view
5. open the note again: last words missing

## Describe what you expected to happen

everything written should remain in the note

**Comment by marreromarco** [comment]
Same issue on Huawei P40 lite E. 
OS specifics: EMUI 10.1.0.167

Furthermore, Sync with Joplin Desktop  is not working properly on Manjaro Linux (AppImage 2.1.7). Notes created on desktop are not synced on Android and vice versa.

**Comment by FireDragonRider** [comment]
This bug still exists.

**Comment by ConradHughes** [comment]
I'm having the same problem (Android 9):

- Last edit (sometimes even just one character) will sometimes disappear on exit from editor.
- It doesn't happen every time.
- It's the editor, not the save-and-sync.
- Workaround that seems fairly reliable is to finish every edit by typing and deleting a character: basically doing something whose disappearance you don't care about.

**Comment by ConradHughes** [comment]
Still happening with Joplin 2.3.4.

**Comment by ConradHughes** [comment]
> Does it also happen if you wait a long time before switching away from the app or locking the phone?

Interesting.  That's not my use pattern so I have not noticed, but will try to remember to test for this.

**Comment by ConradHughes** [comment]
Has recurred during the last month, so still present in 2.4.3 I think.

**Comment by Karob2** [comment]
I've been opening the hamburger menu with the sync button at the bottom to watch the sync progress and wait until it is finished before leaving the app. It's a huge pain but so far I haven't had data loss while doing this.

If we aren't allowed to disable automatic syncing (without also disabling manual syncing), then there really needs to at least be an onscreen indicator that syncing is happening, so the users knows that it is not safe for their data for them to switch away from the app.

**Comment by ConradHughes** [comment]
Yes I always check that menu as well.  I don't leave the app until the sync has finished, but this bug doesn't seem to be related to sync: if I close the editor on android, I can then re-open the note only to discover that the last couple of characters I typed have vanished.  Nowadays I type a few characters of rubbish at the end, then delete them, and sometimes I'll come back to find that those characters have reappeared.  I doesn't happen often, but it definitely still happens.

**Comment by ConradHughes** [comment]
I've just noticed that there's an undo button right beside the back button at the top left of the screen (is this new, or am I just really unobservant?).

Is it possible that this bug comes from undo being triggered by accidental touches when people are aiming for the back button right beside it?  Would it be possible to move the history buttons to the right of the button bar, away from the back button?

**Comment by ConradHughes** [comment]
Just to confirm that this is still present in Joplin 2.6.8: has happened twice in the last couple of days.

**Comment by Karob2** [comment]
Yeah, same for me. It has been infrequent, but still happens in the note title or note body sometimes no matter how much care I take in being sure I've pressed the right buttons and waited for sync to finish.

**Comment by ConradHughes** [comment]
Still here, happened only a couple of days ago, Joplin 2.6.8.

**Comment by ConradHughes** [comment]
Still there (Joplin 2.7.2).

**Comment by ConradHughes** [comment]
Still there (Joplin 2.7.2).

**Comment by ConradHughes** [comment]
Still there (Joplin 2.7.2).

**Comment by ConradHughes** [comment]
Still there (Joplin 2.8.1).

**Comment by ConradHughes** [comment]
Still there (Joplin 2.8.1).

**Comment by personalizedrefrigerator** [comment]
This _might_ be related to https://github.com/laurent22/joplin/issues/6732

**Comment by ConradHughes** [comment]
Still there (Joplin 2.8.1).

**Comment by ConradHughes** [comment]
Still there (Joplin 2.8.1).

**Comment by ConradHughes** [comment]
Still there.

**Comment by ConradHughes** [comment]
Still there.

**Comment by Babber** [comment]
I am also experiencing similar issues on Android 12 with Joplin 2.11.32.

**Comment by hyuri** [comment]
Maybe related to this issue? #13309
